Create Plan your upgrade rollout doc (#6211)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/oie-upgrade-rollout-plan/main/index.md

---
title: Plan your upgrade rollout
---

<ApiLifecycle access="ie" />

Upgrading from Classic Engine to Okta Identity Engine (Identity Engine) involves more than a single click in the Identity Engine Upgrade Hub. Before scheduling your production upgrade, decide how to sequence the upgrade across orgs, how to migrate users to the new flows gradually, and how to handle features that don't carry over to Identity Engine. Okta recommends an iterative approach rather than upgrading everything at once.

---

#### Learning outcome

Build a rollout plan that covers org upgrade sequencing, user segment rollout, and feature replacement for features that aren't supported in Identity Engine.

#### What you need

> * Access to the Admin Console for each org you plan to upgrade

Reviewer: ```suggestion
* Access to the Admin Console for each org that you plan to upgrade
```

## Roll out to user segments

After your org is upgraded, your embedded Sign-In Widget or SDK-based apps don't have to switch all users to Identity Engine flows at once. You can run Classic Engine and Identity Engine flows side by side while you validate, then gradually shift traffic.

The following strategies support a gradual rollout:

| Strategy | How it works | Best for |
| --- | --- | --- |
| Code-based routing | Add conditional logic in your app to direct users to the Classic Engine or Identity Engine flow based on a flag, group membership, or user attribute. | Embedded Sign-In Widget or SDK-based apps |
| Network load balancing | Route traffic to separate app instances—one running the Classic Engine code path, one running the Identity Engine code path. | Large-scale apps with multiple server instances |
| Incremental scaling | Start with a small percentage of users on Identity Engine flows and increase it over time. | Controlled risk reduction in production |

Whichever strategy you choose, follow this sequence:

1. Start with internal users or a small test group.

> 2. Validate sign-in, registration, MFA enrollment, and password recovery end-to-end.

Reviewer: ```suggestion
2. Validate sign-in, registration, MFA enrollment, and password recovery flows end to end.
```

3. Monitor for issues in the System Log and from user feedback.
4. Gradually increase the proportion of users on Identity Engine flows.
5. Remove Classic Engine flow code after all users have been fully migrated and validated.

## Upgrade your Sign-In Widget

> If your app uses an embedded Sign-In Widget, verify that it runs version 5.11.0 or later before you schedule the org upgrade. Earlier versions don't support Identity Engine flows. If you haven't already upgraded the widget, do it now as part of your rollout prep, not at the same time as the org upgrade.

Reviewer: ```suggestion
If your app uses an embedded Sign-In Widget, verify that it runs version 5.11.0 or later before you schedule the org upgrade. Earlier versions don't support Identity Engine flows. If you haven't already upgraded the widget, do it now as part of your rollout prep.
```

See [Update event hook endpoints](https://help.okta.com/okta_help.htm?type=oie&id=identity-engine-upgrade-event-hook-endpoint-update).

## Prepare Terraform for the upgrade

If you use Terraform to manage Okta orgs, prepare before you upgrade:

1. Migrate to the latest version of the [Okta Terraform provider](https://registry.terraform.io/providers/okta/okta).
2. Update your HCL scripts to accommodate syntax changes.
3. Sync the `.tfstate` file using Terraform commands so it stays consistent with the updated scripts.
4. Test the changes on a temporary Okta tenant before applying them in production.

After the upgrade, run `terraform plan -refresh=false` to verify that the script and `.tfstate` file are in sync. The output should read `Resources: 0 added, 0 changed, 0 destroyed`. If changes are detected, update the script or `.tfstate` file to match the current state.

See [Prepare Terraform for upgrade](https://help.okta.com/okta_help.htm?type=oie&id=identity-engine-upgrade-prepare-terraform).

## Replace Integrated Windows Authentication

> Integrated Windows Authentication (IWA) routing rules needs to be removed before you can upgrade. The Identity Engine Upgrade Hub blocks the upgrade if any IWA routing rules exist in your org.

Reviewer: ```suggestion
Remove Integrated Windows Authentication (IWA) routing rules before you upgrade. The Identity Engine Upgrade Hub blocks the upgrade if any IWA routing rules exist in your org.
```

> To prepare:

Reviewer: ```suggestion
To prepare, complete the following steps:
```

* Identify all IWA routing rules in the Admin Console.
* Delete each routing rule before scheduling the upgrade.
* Plan a replacement authentication method. Common replacements are desktop SSO with Okta FastPass or certificate-based authentication.
* After the upgrade and Okta FastPass migration are complete, decommission IWA agents and servers.

See [Delete IWA routing rules](https://help.okta.com/okta_help.htm?type=oie&id=csh-dt-remove-iwa-routing-rules).

## Replace Device Trust

Classic Engine Device Trust isn't supported in Identity Engine. The replacement is Okta Verify with managed certificates and Okta FastPass.

> Start by identifying whether your org uses Device Trust for Desktop, mobile, or both, then plan the migration path for each.

Reviewer: ```suggestion
Start by identifying whether your org uses Device Trust for Desktop, mobile, or both, and then plan the migration path for each.
```

After you [remove the IWA routing rules](#replace-integrated-windows-authentication) and complete the upgrade, migrate desktop devices to Okta FastPass:

1. Configure Device Integration and establish a new Certificate Authority (CA).
2. Import the new CA into your Device Management software.
3. Deploy the CA to all managed devices.
4. Roll out Okta Verify to all devices.
5. Decommission legacy IWA agents and servers.
6. Remove Classic Engine Device Trust platforms.

Mobile device migration is a separate process. See [Device Trust upgrade considerations for desktop](https://help.okta.com/okta_help.htm?type=oie&id=identity-engine-upgrade-dt-upgrade-considerations-desktop) for more details.

## Replace Okta Mobile

> Okta Mobile is deprecated and unavailable after the Identity Engine upgrade. Before upgrading, make sure that all affected users are enrolled in Okta Verify.

Reviewer: ```suggestion
Okta Mobile is unavailable in Identity Engine. Before upgrading, make sure that all affected users are enrolled in Okta Verify.
```

* Identify which users currently rely on Okta Mobile.

> * Communicate the deprecation timeline to those users.

Reviewer: is the phrase "deprecation timeline" accurate? Okay if so, but I don't think we call other features that are Classic-only "deprecated in OIE"

Author: Thank you for flagging that. Okta Mobile went EOL on May 31, 2026. Removed that point.
